src/app/layout.tsx

```diff
@@ -25,11 +29,14 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
-      <body
-        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
-      >
-        {children}
-      </body>
+      <UserProvider>
+        <body
+          className={`${geistSans.variable} ${geistMono.variable} antialiased flex flex-col min-h-screen`}
+        >
+          <main className="flex-grow">{children}</main>
+          <Footer />
```

Comment:
I don't think we need a Footer for pre-auth pages. Since this is the RootLayout, adding the footer here will render in every page. Lets add the Footer only for post-auth pages like `onboarding`

Reply:
we can discuss on this sir if no issue because i think footer should be there and also i think its not required